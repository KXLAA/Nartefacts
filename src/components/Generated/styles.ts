import styled from 'styled-components'

import { getSize, getColor } from '@/styles/global'

export const ImageWrapper = styled.div<{ small?: boolean }>`
  border: ${({ small }) => (small ? '1px' : getSize('xxs'))} solid
    ${getColor('grayLight')};
  border-radius: ${({ small }) => (small ? getSize('xs') : getSize('sm'))};
  padding: 0;

  span {
    border-radius: ${getSize('sm')};
  }
`
