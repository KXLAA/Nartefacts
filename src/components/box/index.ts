export { Box } from './Box'
export type { BoxProps } from './box.types'
