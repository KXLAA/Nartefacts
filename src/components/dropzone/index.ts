export { Dropzone } from './Dropzone'
export type { UploadState } from './types'
