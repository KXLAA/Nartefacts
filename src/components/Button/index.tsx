import * as S from './styles'
import { capitalize } from 'utils/utils'

export type ButtonProps = {
  text?: string
  buttonColor?: string
  textColor?: string
}

export const Button: React.FC<ButtonProps> = ({
  buttonColor,
  text,
  textColor,
}) => {
  return (
    <S.Button buttonColor={buttonColor} textColor={textColor} role="button">
      {capitalize(text!) || 'Button'}
    </S.Button>
  )
}
