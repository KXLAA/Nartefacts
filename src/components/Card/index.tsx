import Image from 'next/image'

import * as S from '@/components/Card/styles'

export type CardProps = {
  image?: string | null
  colors?: string[] | null
  likeCount?: number | null
  textColor?: string | null
  albumArt?: string | null
  spotify?: string | null
  apple?: string | null
  type?: string | null
  description?: string | null
  title?: string | null
  artist?: {
    name?: string | null
    photoUrl?: string | null
    biography?: string | null
  } | null
}

export const Card: React.FC<CardProps> = ({ albumArt }) => {
  return (
    <S.Card>
      <S.Gradient />
      <S.ImageWrapper>
        <Image
          src={albumArt!}
          height={385}
          width={385}
          alt={'album art'}
          layout="responsive"
        />
      </S.ImageWrapper>
    </S.Card>
  )
}
