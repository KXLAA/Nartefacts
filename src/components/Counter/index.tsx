import * as S from './styles'

export type CounterProps = {
  count?: number | undefined | null
}

export const Counter: React.FC<CounterProps> = ({ count }) => {
  return (
    <S.Wrapper>
      <S.Counter>
        <p>{`${count} Gradients Generated`} </p>
      </S.Counter>
    </S.Wrapper>
  )
}
