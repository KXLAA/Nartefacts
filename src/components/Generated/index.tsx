import Image from 'next/image'

import { Spacer } from '@/components/Common/Spacer'
import { ColorsTuple, Palette } from '@/components/Palette'
import * as S from '@/components/Generated/styles'

export type GeneratedProps = {
  imageUrl: null | string
  colors: null | ColorsTuple
  small?: boolean
}

export const Generated: React.FC<GeneratedProps> = ({
  imageUrl,
  colors,
  small,
}) => {
  return (
    <div>
      <S.ImageWrapper small={small}>
        {imageUrl && (
          <Image
            src={imageUrl as string}
            height={800}
            width={800}
            alt={'user uploaded image'}
            layout="responsive"
          />
        )}
      </S.ImageWrapper>
      <Spacer size="sm" />
      <Palette colors={colors as ColorsTuple} small={small} />
    </div>
  )
}
