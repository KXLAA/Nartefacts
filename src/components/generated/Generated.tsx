import Image from 'next/image'

import { ExportPaletteModal } from '@/components/modal'
import { ColorsTuple, Palette } from '@/components/palette'

import { StyledImageWrapper, StyledWrapper } from './generated.styles'
import { GeneratedProps } from './generated.types'

export const Generated = ({ imageUrl, colors, small }: GeneratedProps) => {
  return (
    <StyledWrapper small={small}>
      <StyledImageWrapper small={small}>
        <Image
          src={imageUrl || '/static/placeholder.png'}
          height={800}
          width={800}
          alt={'user uploaded image'}
          layout="responsive"
        />
      </StyledImageWrapper>
      <ExportPaletteModal small={small} colors={colors as ColorsTuple} />
      <Palette
        colors={colors as ColorsTuple}
        size={
          small
            ? 'sm'
            : {
                '@initial': 'md',
                '@md': 'lg',
              }
        }
      />
    </StyledWrapper>
  )
}
