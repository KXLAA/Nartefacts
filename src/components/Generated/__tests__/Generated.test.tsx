import { render } from '@testing-library/react'

import { Generated } from '@/components/Generated'
import { colors, imageUrl } from '@/components/utils'

describe('<Generated />', () => {
  it('should render the uploaded image based on props', () => {
    const { getByRole } = render(
      <Generated colors={colors} imageUrl={imageUrl} />,
    )
    const image = getByRole('img')

    expect(image).toBeTruthy()
  })

  it('should render <Generated /> correctly', () => {
    const { container } = render(
      <Generated colors={colors} imageUrl={imageUrl} />,
    )
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with image placeholder if imageUrl is invalid  ', () => {
    const { getByRole } = render(<Generated colors={colors} imageUrl={''} />)
    const image = getByRole('img')
    expect(image).toBeTruthy()
  })
})
