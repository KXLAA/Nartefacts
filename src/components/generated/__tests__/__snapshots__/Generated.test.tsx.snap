// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Generated /> should render <Generated /> correctly 1`] = `
<div
  class="nartefacts-c-PJLV nartefacts-c-bMThXb"
>
  <div
    class="nartefacts-c-lmYeeW"
  >
    <span
      style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 100% 0px 0px 0px;"
      />
      <img
        alt="user uploaded image"
        data-nimg="responsive"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(https://ucarecdn.com/c515e4b5-a5bb-44a3-85d0-f139497e0de0/forbrokenearsep_tems.jpeg); background-position: 0% 0%;"
      />
      <noscript />
    </span>
  </div>
  <button
    aria-controls="radix-3"
    aria-expanded="false"
    aria-haspopup="dialog"
    class="nartefacts-c-hctSdv nartefacts-c-hctSdv-ksKRLf-size-sm nartefacts-c-hctSdv-eVtbBr-variant-dark nartefacts-c-hctSdv-deuXAd-fullWidth-true nartefacts-c-hctSdv-etFzOM-size-md nartefacts-c-hctSdv-iPJLV-css"
    data-state="closed"
    data-testid="export-button"
    type="button"
  >
    export
  </button>
  <div>
    <div
      class="react-horizontal-scrolling-menu--wrapper "
    >
      <div
        class="react-horizontal-scrolling-menu--header"
      />
      <div
        class="react-horizontal-scrolling-menu--inner-wrapper"
      >
        <div
          class="react-horizontal-scrolling-menu--arrow-left"
        />
        <div
          class="react-horizontal-scrolling-menu--scroll-container "
        >
          <div
            class="react-horizontal-scrolling-menu--item "
            data-index="0"
            data-key="#095525"
          >
            <div
              class="nartefacts-c-PJLV nartefacts-c-DxlRh nartefacts-c-DxlRh-uAklm-size-md nartefacts-c-DxlRh-dhVnEv-size-lg nartefacts-c-PJLV-icndjIz-css"
              data-testid="color-box"
            >
              <span />
            </div>
          </div>
          <div
            class="react-horizontal-scrolling-menu--separator "
            data-index="0.1"
            data-key="#095525-separator"
          />
          <div
            class="react-horizontal-scrolling-menu--item "
            data-index="1"
            data-key="#673636"
          >
            <div
              class="nartefacts-c-PJLV nartefacts-c-DxlRh nartefacts-c-DxlRh-uAklm-size-md nartefacts-c-DxlRh-dhVnEv-size-lg nartefacts-c-PJLV-ikUUhWU-css"
              data-testid="color-box"
            >
              <span />
            </div>
          </div>
          <div
            class="react-horizontal-scrolling-menu--separator "
            data-index="1.1"
            data-key="#673636-separator"
          />
          <div
            class="react-horizontal-scrolling-menu--item "
            data-index="2"
            data-key="#379273"
          >
            <div
              class="nartefacts-c-PJLV nartefacts-c-DxlRh nartefacts-c-DxlRh-uAklm-size-md nartefacts-c-DxlRh-dhVnEv-size-lg nartefacts-c-PJLV-ijASQqu-css"
              data-testid="color-box"
            >
              <span />
            </div>
          </div>
          <div
            class="react-horizontal-scrolling-menu--separator "
            data-index="2.1"
            data-key="#379273-separator"
          />
          <div
            class="react-horizontal-scrolling-menu--item "
            data-index="3"
            data-key="#136407"
          >
            <div
              class="nartefacts-c-PJLV nartefacts-c-DxlRh nartefacts-c-DxlRh-uAklm-size-md nartefacts-c-DxlRh-dhVnEv-size-lg nartefacts-c-PJLV-ibBDuEG-css"
              data-testid="color-box"
            >
              <span />
            </div>
          </div>
          <div
            class="react-horizontal-scrolling-menu--separator "
            data-index="3.1"
            data-key="#136407-separator"
          />
          <div
            class="react-horizontal-scrolling-menu--item "
            data-index="4"
            data-key="#573227"
          >
            <div
              class="nartefacts-c-PJLV nartefacts-c-DxlRh nartefacts-c-DxlRh-uAklm-size-md nartefacts-c-DxlRh-dhVnEv-size-lg nartefacts-c-PJLV-igZxaNn-css"
              data-testid="color-box"
            >
              <span />
            </div>
          </div>
          <div
            class="react-horizontal-scrolling-menu--separator "
            data-index="4.1"
            data-key="#573227-separator"
          />
          <div
            class="react-horizontal-scrolling-menu--item "
            data-index="5"
            data-key="#547820"
          >
            <div
              class="nartefacts-c-PJLV nartefacts-c-DxlRh nartefacts-c-DxlRh-uAklm-size-md nartefacts-c-DxlRh-dhVnEv-size-lg nartefacts-c-PJLV-ifyEfth-css"
              data-testid="color-box"
            >
              <span />
            </div>
          </div>
          <div
            class="react-horizontal-scrolling-menu--separator "
            data-index="5.1"
            data-key="#547820-separator"
          />
          <div
            class="react-horizontal-scrolling-menu--item "
            data-index="6"
            data-key="#163598"
          >
            <div
              class="nartefacts-c-PJLV nartefacts-c-DxlRh nartefacts-c-DxlRh-uAklm-size-md nartefacts-c-DxlRh-dhVnEv-size-lg nartefacts-c-PJLV-iihcxYh-css"
              data-testid="color-box"
            >
              <span />
            </div>
          </div>
          <div
            class="react-horizontal-scrolling-menu--separator "
            data-index="6.1"
            data-key="#163598-separator"
          />
          <div
            class="react-horizontal-scrolling-menu--item "
            data-index="7"
            data-key="#682052"
          >
            <div
              class="nartefacts-c-PJLV nartefacts-c-DxlRh nartefacts-c-DxlRh-uAklm-size-md nartefacts-c-DxlRh-dhVnEv-size-lg nartefacts-c-PJLV-ifUsSfW-css"
              data-testid="color-box"
            >
              <span />
            </div>
          </div>
        </div>
        <div
          class="react-horizontal-scrolling-menu--arrow-right"
        />
      </div>
      <div
        class="react-horizontal-scrolling-menu--footer"
      />
    </div>
  </div>
</div>
`;

exports[`<Generated /> should render with image placeholder if imageUrl is invalid   1`] = `
<img
  alt="user uploaded image"
  data-nimg="responsive"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; filter: blur(20px); background-size: cover; background-image: url(/static/placeholder.png); background-position: 0% 0%;"
/>
`;
